### PR TITLE
Issue #SB-23774 feat: Redesign the entire Groups instrumentation

### PR DESCRIPTION
**Pull request by kamaleeswary**

### CODE REVIEW

**Review by VenkatKadiveti — APPROVED** [review]

File: src/app/client/src/app/modules/groups/components/group-header/group-header.component.ts [hunk 1]
```diff
@@ -97,21 +97,24 @@ export class GroupHeaderComponent implements OnInit, OnDestroy {
...
-
   toggleModal(visibility = false, name?: string, eventName?: string) {
     this.showModal = visibility;
     this.groupService.emitMenuVisibility('group');
     this.name = name;
     switch (name) {
       case actions.DELETE:
-        this.addTelemetry('delete-group', {status: _.get(this.groupData, 'status')});
+        this.addTelemetry('delete-group', {status: _.get(this.groupData, 'status')}, {type: 'select-delete'});
         this.assignModalStrings(this.resourceService.frmelmnts.lbl.deleteGroup, this.resourceService.messages.imsg.m0082, '{groupName}');
```
Comment (sudip-mukherjee) [inline src/app/client/src/app/modules/groups/components/group-header/group-header.component.ts:114]: here as well. check the types u added. FYI @vinukumar-vs

**Review by sudip-mukherjee — CHANGES_REQUESTED** [review]

File: src/app/client/src/app/modules/groups/components/group-header/group-header.component.ts (hunk 1 above)
Comment (kamaleeswary) [reply src/app/client/src/app/modules/groups/components/group-header/group-header.component.ts:114]: type is given by sai CC: @vinukumar-vs

File: src/app/client/src/app/modules/groups/components/create-edit-group/create-edit-group.component.spec.ts [hunk 2]
```diff
@@ -155,7 +155,7 @@ describe('CreateEditGroupComponent', () => {
     spyOn(component['groupService'], 'addTelemetry');
     component.groupId = '123';
     component.addTelemetry('ftu-popup');
-    expect(component['groupService'].addTelemetry).toHaveBeenCalledWith({id: 'ftu-popup', extra: undefined},
+    expect(component['groupService'].addTelemetry).toHaveBeenCalledWith({id: 'ftu-popup', extra: undefined, edata: undefined },
     fakeActivatedRouteWithGroupId.snapshot, [], '123');
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/client/src/app/modules/groups/components/create-edit-group/create-edit-group.component.spec.ts:159]: I will change this

File: src/app/client/src/app/modules/groups/components/create-edit-group/create-edit-group.component.ts [hunk 3]
```diff
@@ -136,10 +154,37 @@ export class CreateEditGroupComponent implements OnInit, OnDestroy {
...
+    this.groupService.addTelemetry({id, extra, edata}, this.activatedRoute.snapshot, [], this.groupId);
   }
 
+  setTelemetryImpression(edata?) {
+    this.telemetryImpression = {
+      context: {
+        env: this.activatedRoute.snapshot.data.telemetry.env,
+        cdata: [{
+          type: 'Group',
+          id: ''
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/client/src/app/modules/groups/components/create-edit-group/create-edit-group.component.ts:171]: when we are creating the group id will not be there so

File: src/app/client/src/app/modules/groups/components/add-member/add-member.component.html [hunk 4]
```diff
@@ -30,7 +30,7 @@
               </div>
             </div>
             <re-captcha class="recaptcha" *ngIf="isCaptchEnabled" #captchaRef="reCaptcha" (resolved)="$event && captchaResolved($event)" siteKey="{{googleCaptchaSiteKey}}" size="invisible"></re-captcha>
-            <button type="button" class="sb-btn sb-btn-normal sb-btn-nolayer" [ngClass]="{'sb-btn-primary': memberId?.trim(), 'sb-btn-disabled': !memberId?.trim()}" [disabled]="!memberId?.trim()" (click)="onVerifyMember();setInteractData('verify-member', {searchQuery: memberId})">{{resourceService?.frmelmnts?.btn?.verify}}</button>
+            <button type="button" class="sb-btn sb-btn-normal sb-btn-nolayer" [ngClass]="{'sb-btn-primary': memberId?.trim(), 'sb-btn-disabled': !memberId?.trim()}" [disabled]="!memberId?.trim()" (click)="onVerifyMember();setInteractData('verify-member', {searchQuery: memberId}, '', {type: 'verify-user'})">{{resourceService?.frmelmnts?.btn?.verify}}</button>
           </div>
```
[the remark this answers is not in the document]
Comment (kamaleeswary) [reply src/app/client/src/app/modules/groups/components/add-member/add-member.component.html:34]: we can't use the id as a type. The first one is for id and the next is for type. type and id will not be the same for all the interact events.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]
Use single telemeryContstant file for all the labels used for telemetry. 
So that later any changes or use of same name can be referred from the constant file object.

**Comment by kamaleeswary** [comment]
> Use single telemeryContstant file for all the labels used for telemetry.
> So that later any changes or use of same name can be referred from the constant file object.

**Review by vinukumar-vs — APPROVED** [review]

**Pull request merged**